File: BaseballApi/Models/Fielder.cs
namespace BaseballApi;

public class Fielder
{
    public long Id { get; set; }

    public long BoxScoreId { get; set; }
    public required BoxScore BoxScore { get; set; }

    public long PlayerId { get; set; }
    public required Player Player { get; set; }

    public int Number { get; set; }
    public int Games { get; set; }
    public int Errors { get; set; }
    public int ErrorsThrowing { get; set; }
    public int ErrorsFielding { get; set; }
    public int Putouts { get; set; }
    public int Assists { get; set; }
    public int StolenBaseAttempts { get; set; }
    public int CaughtStealing { get; set; }
    public int DoublePlays { get; set; }
    public int TriplePlays { get; set; }
    public int PassedBalls { get; set; }
    public int PickoffFailed { get; set; }
    public int PickoffSuccess { get; set; }
}

File: iScoreImport/iScoreImport/Models/Fielder.swift
//
//  Fielder.swift
//  iScoreImport
//
//  Created by Patrick Eschenfeldt on 7/4/24.
//

import Foundation

struct Fielder: Codable {
    let PlayerExternalId: UUID
    var Player: Player?
    let TeamExternalId: UUID
    
    let Number: Int
    let Games: Int
    let ErrorsThrowing: Int
    let ErrorsFielding: Int
    let Putouts: Int
    let Assists: Int
    let StolenBaseAttempts: Int
    let CaughtStealing: Int
    let DoublePlays: Int
    let TriplePlays: Int
    let PassedBalls: Int
    let PickoffFailed: Int
    let PickoffSuccess: Int
}
